### DISCUSSION TITLE
General Questions and Possible Bug (E3V2-BLTouch-5x5-HS-v4.2.2)

### DISCUSSION THREAD

**Discussion by poorandunlucky** [Jyers/Marlin#657] [Q&A] [answered]
### So the story is:

I got a Ender 3 V2 for the nice screen for my S/O so he could use it easily, but then when I tried to compiled Marlin, I found out about the DWIN screen, and how I couldn't get most of the options, and the stock firmware wouldn't play nicely with my ANCLABS BLTOUCH 3.1 that I got from Creality themselves, the needle would get stuck, wouldn't complete a 3x3, the offsets were all wrong, [...] needless to say Creality should pay you handsomely for your firmware.

I was about to get a standard 128 x 64 LCD display since I found out I could just use the same cable, and it would be an easy swap, but looking I found your firmware, and decided to try it.  I'm _very_ impressed by your work!  **Very!**

BUT!  I ran into a few hicks, and I would like to consult with some people more experienced than I am about a few details I might've gotten wrong... it's easy these days with the information overflow, and the overabundance of people willing to share what they think they know (with mixed results).

### Anyway, the point:

The firmware works fine, it levels 5x5, I set the probe offset to -45/-4.5/-3, I mention it because I imagine most BLTOUCH are about these offsets, so just checking, but it prints.  I chose 5x5 because of past problems with the probe, I wanted to make sure it would work before going with UBL.

The problem is that when it prints, it doesn't stay on the printing screen, it goes back to the main menu, and I can't control the print (pause/tweak/stop/etc.).  If I want to stop, like I was testing with a torus, and I forgot the glue, and I had to power it off.

Edit: The print finished while writing this, I just took it off, and when I did, the screen was at the normal "Print Completed" screen, like the screen you normally get saying it's done, with the time it took, and you have to click to acknowledge, and return to the main menu.

When the print starts, I have the normal status screen with Tweak/Pause/Stop, but along the way, it goes back to the main menu where I can't control anything, and when it ends, it goes back to the screen it's supposed to be on...  So after clicking a file, it goes to the Print Status Screen, it warms-up, it goes back to the main menu while printing, the print finishes, and it goes back to the Print Status Screen saying the print is done.

**So that's the possible bug.**

### The question is:

Do I need to add M501 in my start G-Code in Cura to retrieve the bed leveling information?  Is it best to level the bed at every print (G29) [I never used GCode directly, I'm not sure I'm getting this right]?

Now it's printing a cube quite nicely, and I left the Cura GCode defaults for the Ender 3 Pro.  Aside from that start, and stop GCode, and the GCode for the cube itself, everything is default.

In the firmware, on the screen, I ran the bed leveling, I saved it, I changed some colors, I ran the autotune for the nozzle, and hotbed, I set the offsets, I changed my preheat to 150/50, and that's it.

I **did not** update the screen firmware, I am not sure if I can just take the DWIN folder in the repo, and flash it with that, or if I have to run the scripts or whatnot, I'm not sure what to do, so I left it alone...  Should I flash the screen, do I just put the DWIN .bin, .cfg, and .ico straight from the repo, put it on the SD and flash it with that?

### Also:

When I switch to UBL, is there anything else I need to do?  Do I need to add GCode anywhere, will it do its thing automatically, I just read on the Marlin website that it did M501 on reboot, but I just want to be certain when I switch to UBL of what's different, what I need to do, I'd just like to get my bearings, because it's been a real disappointment with the Creality firmware, and the DWIN screen, and compiling Marlin was straightforward, but in the end, I had to slim my build down so much it I was better off with the Creality one, plus the probe wouldn't work properly...

Aside from going back to main menu during printing (I can take a video if you like, if it helps), and though I'm not sure if it's even using the leveling data, or just printing willy-nilly, your firmware is **Magic**!  I am not sure I could ever thank you enough.  Even as it is, it's a whole lot better than everything I've tried so far...  So many "thank yous" from the bottom of my heart, it's really ... you deserve a freakin' plaque or something!

### In conclusion:

In exchange for your help, I would be happy to write a "Getting Started From Defaults With Precompiled Releases" page for your Wiki, if you want...  So that someone else like me, in the future, someone confused by a lot of trial and error, can just use those, start from defaults, and know what to do, where they are, and what's happening...  As you can see, I can ... write.  A lot.  So...  I'd be happy to help in exchange  😊

### Other Small Question:

Would there be a way to have it prehead on boot?  Like when it boots, to have the bed come to a "standby" temperature, and so stay at that temperature until maybe a timeout of 5 minutes?  Say it boots, and immediately starts going to 130/40 so when you're ready to print, it doesn't take ages to come to temperature, and so it doesn't go down to 0 between prints?  Having to stand there with the glue stick for it to come to temperature so I can apply it just before the print is really unpleasant...

I don't know if it's possible this way, or if this is a feature request for you, or for Marlin, or if I could do it with GCode or something, but I think it would be a nice option...

**Comment by Cal-LF** [comment]
Note: questions with many paragraphs are hard to follow.

Mine was jumping to the  tune menu by itself. It is not a bug, I did the grounding fix by Creality and the issue stopped. 
Also make sure the printer is using a grounded outlet and circuit.

![Image02](https://user-images.githubusercontent.com/76499230/116787736-dcf5f680-aa73-11eb-9765-913788ceeee8.jpg)

**Comment by poorandunlucky** [comment, marked as answer]
OK, so I bit the bullet, and flased the screen, and I also found the M420 S1 / G29 A G29 J in the wiki...

I also flashed the 1.3.2 fw because, well, why not?  Lol

It's printing now, and the screen is behaving normally.  I should be OK...

Thanks to you both  😁

**Comment by Jyers** [comment]
As far as the print screen disappearing. This is a known issue and I'll solve it soon. It occurs if you have M501 in your start gcode. For now simply removing it should solve the issue.

On the topic of the wiki, that would be wonderful! Feel free to email me with a draft or any questions about it at Jyers.Marlin@gmail.com
